docs/design/pitr-mysql.md

```diff
@@ -0,0 +1,218 @@
+- Check that no connection exists on the original database
+
+## Crash
+If the recovery process crashes before completion, we should mark the recovery as failed. The user should start a new recovery process.
+
+
+# API and Schema
+
+Bytebase should make some changes to the API and schema design to implement the PITR feature.
+
+## Backup PITR Information
+
+When taking a backup of a database, now we also need to store the current position in the data change stream when the logical backup is dumped. In MySQL, this position refers to binary log file name and position. In PostgreSQL, it’s the WAL lsn. To use a single model, we store a JSON encoded string that could be encoded and parsed, respectively.
+
+To provide a search option with the timestamp and binlog position, we will encode both of them, where the timestamp is the time at which the statement began executing on the MySQL server and decoded from the [binlog event header](https://dev.mysql.com/doc/internals/en/event-header-fields.html). The timestamp is represented as the UNIX timestamp, i.e., the number of seconds since 1970 (UTC).
+
+An example of the PITR information is like this:
+
+```json
+{
+    "binlog_info": {
+        "binlog_name": "binlog.000001",
+        "binlog_position": 1234,
+        "created_ts": 1650957790
+    }
+}
+```
+
+We will add a field in api.Backup:
+
+```go
+type Backup struct {
+	...
+	// Payload contains arbitrary string message with following cases
+	// 1. contains the starting position of incremental backup when this backup snapshot is taken.
+	// This field is not returned to the frontend.
+	Payload string
+}
+```
+
+We will add a column in table backup:
+
+```sql
+CREATE TABLE backup (
+    ...
+    payload TEXT NOT NULL
+);
+```
+
+When the user performs a PITR recovery, Bytebase will check the api.Backup and try to find the PITR information. If the PITR information is not empty, Bytebase will perform a binlog replay after recovering the logical backup using the [recovery process described above](#recovery).
+
+
+# Data Retention Policy
+
+As time goes by, the logical backup and incremental backup will take up infinite storage space, and a data retention policy should be defined to solve this problem.
+
+Bytebase will currently implement a default data retention policy of 7 days. We may later make this configurable as an improvement.
+
+
+# Monitoring
+
+Steps 3 and 4 could take some time, and Bytebase should export the progress via api.Task, which could be obtained from an api.Issue.
+
+
+# Third-Party Integration
+
+Bytebase should provide event webhooks for user system integration. For example, we could send the logical backup success/failure events to a user-defined URL.
+
+
+# Test Plan
+
+To test the Point-in-time Recovery is working as expected, we will write several unit test cases following the [Critical User Journeys](#critical-user-journeys) described above. The test cases should cover major path and corner cases as possible.
```

Comment:
Normal case, perform a PITR
Advanced case, perform a PITR on the database restored from a PITR